### PR TITLE
CP Subsystem - CountDownLatch

### PR DESCRIPTION
**Pull request by mdumandag**
Adds the implementation, documentation, code samples, and test of
CP CountDownLatch.

Also, the invocation service is edited to return `OperationTimeoutError`
instead of `HazelcastTimeoutError` in case of invocation timeouts
as the Java client does.

depends on #224

### CODE REVIEW

**Comment by mdumandag** [comment]
verify

File: tests/proxy/cp/count_down_latch_test.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        self.assertEqual(0, latch.get_count())
+        latch.try_set_count(10)
+        self.assertEqual(10, latch.get_count())
+
+    def test_try_set_count(self):
+        latch = self._get_latch()
+        self.assertTrue(latch.try_set_count(3))
+        self.assertEqual(3, latch.get_count())
+
+    def test_try_set_count_when_count_is_already_set(self):
```
Comment (puzpuzpuz) [inline tests/proxy/cp/count_down_latch_test.py:135]: nit: does it make sense include a test that verifies that `try_set_count` succeeds once the count goes to zero?

**Review by puzpuzpuz — APPROVED** [review]
LGTM

File: tests/proxy/cp/count_down_latch_test.py (hunk 1 above)
Comment (mdumandag) [reply tests/proxy/cp/count_down_latch_test.py:135]: Good idea. I added the extra test case.

**Comment by mdumandag** [comment]
Thanks for the review Andrey, I am merging this PR
